libyul/AsmAnalysis.h

```diff
@@ -60,20 +60,30 @@ class AsmAnalyzer
 		AsmAnalysisInfo& _analysisInfo,
 		langutil::ErrorReporter& _errorReporter,
 		Dialect const& _dialect,
+		std::string const& _objectName,
 		ExternalIdentifierAccess::Resolver _resolver = ExternalIdentifierAccess::Resolver(),
 		std::set<std::string> _dataNames = {}
 	):
 		m_resolver(std::move(_resolver)),
 		m_info(_analysisInfo),
 		m_errorReporter(_errorReporter),
 		m_dialect(_dialect),
-		m_dataNames(std::move(_dataNames))
+		m_dataNames(std::move(_dataNames)),
+		m_objectName(std::move(_objectName))
 	{
 		if (EVMDialect const* evmDialect = dynamic_cast<EVMDialect const*>(&m_dialect))
 		{
 			m_evmVersion = evmDialect->evmVersion();
 			m_eofVersion = evmDialect->eofVersion();
 		}
+
+		if (m_eofVersion) // No dots in object name for EOF
+		{
+			for (auto const& name: m_dataNames)
+				solAssert(name.find('.') == std::string::npos, "Dot character not allowed in object name for EOF");
+
+			solAssert(m_objectName.find('.') == std::string::npos, "Dot character not allowed in object name for EOF");
```

Comment:
This one must be reported a regular error, not an assert/panic, since it can normally occur in the code. Please use `m_errorReporter.syntaxError()`.

As a regular error, this check must be moved to `analyze()`. The constructor is too early. I'd actually add a new function to group all checks related to object structure (e.g. `validateObjectStructure(_objectName, _qualifiedDataNames)`) and call that from the `try` block in `analyze()`.

Also, you can only enforce it for `m_objectName`. `m_dataNames` will still contain dot-separated paths, even on EOF, it's just that now you'll be able to safely assume that you can split them on dots if needed. I mean, we could also change that and start passing only the immediate data subobjects, because on EOF you cannot refer to objects more than one level deep, but I'd rather keep that part of the interface of `AsmAnalyzer` as is, not to complicate its use.

Reply:
As for the arguments of `eofcreate()` and `returncontract()` - those may or may not be validated in #15512. If you don't do it, paths will still be rejected, because they won't ever match a name. But having a specific error saying that the values are invalid because we expect names, not paths would be nicer for the user.

Reply:
And if we `qualifiedDataNames` we can also do not add object name to the set and the change with passing opbjectName to the AsmAnalysis is not needed any longer.

Reply:
Or maybe not because we wanna check that the object name does not contain dot

Reply:
Yeah, nothing changes in what you pass in. Just in how you validate it.